Wi-Fi shell improvements

krish2718 opened the pull request.

krish2718 committed:
wifi: shell: Fix the arg count for reg domain

Missed accounting for "-f" option.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

krish2718 committed:
wifi: shell: Fix optional arg count for connect

Fix an extra optional arg.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

krish2718 committed:
wifi: shell: Fix help for PS command

Clearly mark the args as optional.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

krish2718 committed:
wifi: shell: Remove the unnecessary text in scan

We are using standard notation to differentiate optional and mandatory,
so, no need for a heading.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

krish2718 committed:
wifi: shell: Fix the help for reg domain

Separate the two optional parameters and add help.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

krish2718 committed:
wifi: shell: Remove the unnecessary text

The parameters heading is implied and doesn't have the newline, so, just
remove it.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

krish2718 committed:
wifi: shell: Fix the inconsistency in commands separation

For better readability, below rules will help:

 * Each command should be separated by a newline
 * Each command should end with a full stop (intermediate statements
   shouldn't have full stops)

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

krish2718 committed:
shell: Add a space after colon

Just a cosmetic change, but IMHO the help looks much better now :).

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

sachinthegreen commented on subsys/net/l2/wifi/wifi_shell.c:
```diff
@@ -1651,111 +1651,108 @@ SHELL_STATIC_SUBCMD_SET_CREATE(wifi_commands,
 		  cmd_wifi_disconnect,
 		  1, 0),
-	SHELL_CMD_ARG(ps, NULL, "Configure Wi-F PS on/off, no arguments will dump config",
+	SHELL_CMD_ARG(ps, NULL, "Configure Wi-F power save state.\n"
```
```suggestion
	SHELL_CMD_ARG(ps, NULL, "Configure or display Wi-Fi power save state.\n"
```

krish2718 commented:
FYI, for a single line change in help, had to pull in many of the `scan` re-work `fromtree` commits, as upmerge is still pending.

jukkar commented:
> FYI, for a single line change in help, had to pull in many of the `scan` re-work `fromtree` commits, as upmerge is still pending.

Just wondering if you could delay the related ncs work until the upstream patch is merged, and only then do downstream commits etc.

krish2718 commented:
> > FYI, for a single line change in help, had to pull in many of the `scan` re-work `fromtree` commits, as upmerge is still pending.
> 
> Just wondering if you could delay the related ncs work until the upstream patch is merged, and only then do downstream commits etc.

The reason for doing this in tandem it to run the NCS CI before upstream PR is merged, this way we can catch any bugs easily. And once its merged, we can abandon the PR or merge it.

> FYI, for a single line change in help, had to pull in many of the `scan` re-work `fromtree` commits, as upmerge is still pending.

Just realized that this comment was supposed to be in downstream :-).

krish2718 committed:
wifi: shell: Use case insensitive comparison

The help text uses the capital case as its an acronym, but passing
capital case fails. Also extend that to others as well.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

rlubos reviewed (dismissed):
Looks good, but please check the twister results. Looks like some include is missing? At least  minimal libc declares `strncasecmp()` in `strings.h`.
